src/compiler/transformer.ts

```diff
@@ -664,12 +666,10 @@ export function transformNodes<T extends Node>(resolver: EmitResolver | undefine
 }
 
 /** @internal */
-export const nullTransformationContext: TransformationContext = {
+export const nullTransformationContext: NullTransformationContext = {
+    kind: TransformationContextKind.NullContext,
     factory: factory, // eslint-disable-line object-shorthand
     getCompilerOptions: () => ({}),
-    getEmitResolver: notImplemented,

```

Comment:
Since some of the removed members throw an exception at runtime, I would not expose them on the `NullTransformationContext` type. If someone somewhere is actually using the `nullTransformationContext` as a full transformation context I kept the `nullTransformationContext` to have the same runtime members but hid it behind the more restrictive `NullTransformationContext`

Or did you have in mind that the `NullTransformationContext` type is more similar to a full `TransformationContext`. I don't particularly like that idea since some of them throw runtime exceptions so ideally the type system should not let you access those members.

Reply:
`NullTransformationContext` has always just been a normal `TransformationContext` with the features that depended on a type checker or program disabled. Having the shape remain stable avoids V8 recompiling large chunks of tsc.js because the checker sees one shape when it creates synthetic nodes when producing diagnostics via `typeToTypeNode` and another shape when it sees a full checker when doing emit.

Reply:
> I don't particularly like that idea since some of them throw runtime exceptions so ideally the type system should not let you access those members.

If we want the type to follow the shape but limit access, we could always define a `NullTransformationContext` type that makes the return types of those methods `never`.

Reply:
@rbuckton I changed `NullTransformationContext` to have the same shape as `TransformationContext` (just the not implemented methods return never)